iwyu_output.cc

```diff
@@ -2298,4 +2306,19 @@ size_t IwyuFileInfo::CalculateAndReportIwyuViolations() {
   return num_edits;
 }
 
+bool IsDirectlyIncluded(const NamedDecl* decl, const IwyuFileInfo& includer) {
+  const vector<string> mapped_headers = internal::GetMappedPublicHeaders(
+      internal::GetQualifiedNameAsString(decl),
+      GetFilePath(includer.file_entry()), GetFilePath(decl));
+  if (!mapped_headers.empty()) {
+    auto pred = [&direct_includes =
+                     includer.direct_includes()](const string& header) {
+      return direct_includes.count(header);
+    };
+    return find_if(mapped_headers.begin(), mapped_headers.end(), pred) !=
```

Comment:
You could consider `llvm::find_if` from `llvm/STLExtras.h` instead; it takes the container/range instead of individual iterators. Not sure if it makes the expression fit on one line, but it might.

Reply:
Nice, thanks! Probably, it would be possible to upgrade C++ standard version used up to C++20 in order to use the standard range algorithms?

Reply:
I've just followed LLVM's lead to avoid surprises (and put less pressure on user toolchains), their lowest supported version is C++17